src/DropdownReferenceSet/components/DropdownReferenceSet.ts

```diff
@@ -0,0 +1,153 @@
+            const dropdown = document.getElementsByClassName("Select-menu-outer");
+            scrollContainer.addEventListener("scroll", () => {
+                dropdown[0] ? (dropdown[0] as HTMLElement).style.visibility = "hidden" : window.logger.warn("Dropdown not available");
+                const activeElement = document.activeElement;
+                if (activeElement) {
+                    (activeElement as HTMLElement).blur();
+                }
+            });
+        }
+    }
+
+    private renderSelector() {
+        const commonProps = {
+            clearable: this.props.isClearable,
+            multi: true,
+            removeSelected: true,
+            disabled: this.props.isReadOnly,
+            onChange: this.props.handleOnchange,
+            ...this.createSelectorProp()
+        };
+
+        if (this.props.readOnlyStyle === "control" || (this.props.readOnlyStyle === "text" && !this.props.isReadOnly)) {
+            return createElement("div", {
+                className: classNames("widget-dropdown-reference-set"),
+                onClick: this.setDropdownSize
+            },
+                this.props.selectType === "normal"
+                    ? this.props.isReadOnly
+                        ? createElement("input", {
+                            type: "text",
+                            readonly: "readonly",
+                            className: "form-control",
+                            disabled: "disabled",
+                            value: this.processOptions()
+                        })
+                        : createElement(Select, {
+                            options: this.props.data,
+                            noResultsText: "",
+                            ...commonProps
+                        })
+                    : createElement(Async, {
+                        searchPromptText: this.props.minimumCharacter > 0
+                            ? `Type more than ${this.props.minimumCharacter} characters to search`
+                            : "Type to search",
+                        loadOptions: this.props.asyncData,
+                        ...commonProps
+                    }),
+                createElement(Alert, { className: "widget-dropdown-reference-set-alert" }, this.props.alertMessage)
+            );
+        } else {
+            return createElement("p", { className: classNames("form-control-static", "read-only-text") },
+                this.processOptions()
+            );
+        }
+    }
+
+    private setDropdownSize = () => {
+        const dropdown = document.getElementsByClassName("Select-menu-outer");
+        if (dropdown[0] && (dropdown[0] as HTMLElement).style.visibility !== "visible") {
+            const dropdownDimensions = dropdown[0].getBoundingClientRect();
+            if (dropdown && dropdown.length && dropdownDimensions) {
+                (dropdown[0] as HTMLElement).style.width = dropdownDimensions.width - .08 + "px";
```

Comment:
Use a const instead of use `(dropdown[0] as HTMLElement)` each single line.

eg.
`const dropdownElement = dropdown[0] as HTMLElement;
                dropdownElement.style.width = dropdownDimensions.width - .08 + "px";
                dropdownElement.style.left = dropdownDimensions.left + "px";
                dropdownElement.style.top = dropdownDimensions.top + "px";
                dropdownElement.style.visibility = "visible";
                dropdownElement.style.position = "fixed";`

Reply:
Fixed